scale/queue/job_exe.py

```diff
@@ -61,6 +62,9 @@ def create_job_exe_model(self, framework_id, when):
 
         job_exe.set_cluster_id(framework_id, self._queue.job_id, self._queue.exe_num)
 
+        if self.required_resources.gpus > 0:
+            GPUManager.assign_gpus_for_job(job_exe.node_id,job_exe.job_id, self.required_resources.gpus)
```

Comment:
Check if false is returned, possibly cancel execution? Set as failed?

Reply:
this is just to limit calling assign_gpus_for_job to jobs that actually use GPUs. create_job_exe_model will be called by every job created, and most will not need to interact with the GPUmanager.

Reply:
Right, I was wondering what we should do if the GPU manager is unable to assign gpus. I guess the job just won't be scheduled and will sit in the queue until it is canceled?  Or it will try to schedule the job and it will fail when put on a node with insufficient resources?  I have a PR for reporting such situations, FYI.

Reply:
i got ya. theoretically it shouldn't happen, as scale shouldnt assign more GPUs than are available on a node. if that happens we should probably fail the job.

Reply:
added a check in scheduling_node when GPUs are first reserved so this shouldn't ever happen. just the same I put in a logging message if it ever does.